src/opnsense/mvc/app/controllers/OPNsense/IPsec/Api/LegacySubsystemController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/*
+ * Copyright (C) 2019 Pascal Mathis <mail@pascalmathis.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\IPsec\Api;
+
+use OPNsense\Base\ApiControllerBase;
+
+require_once('guiconfig.inc');
+require_once('system.inc');
+require_once('filter.inc');
+require_once('services.inc');
+require_once('interfaces.inc');
+require_once('plugins.inc.d/ipsec.inc');
+
+/**
+ * Class LegacySubsystemController
+ * @package OPNsense\IPsec\Api
+ */
+class LegacySubsystemController extends ApiControllerBase
+{
+    /**
+     * Returns the status of the legacy subsystem, which currently only includes a boolean specifying if the subsystem
+     * is marked as dirty, which means that there are pending changes.
+     * @return array
+     */
+    public function statusAction()
+    {
+        return ['isDirty' => is_subsystem_dirty('ipsec')];
+    }
+
+    /**
+     * Apply the IPsec configuration using the legacy subsystem and return a message describing the result
+     * @return array
+     */
+    public function applyConfigAction()
+    {
+        if ($this->request->isPost()) {
```

Comment:
Since the code to call doesn't have a new alternative, isolate the logic and call via configd.

Very likely it would already be sufficient to call `pluginctl -c ipsec` and remove the "dirty" file when done. (the filter shouldn't be necessary) 

add action to https://github.com/opnsense/core/blob/master/src/opnsense/service/conf/actions.d/actions_ipsec.conf

then call the usual way, like 
https://github.com/opnsense/core/blob/5689dc8048cfe739f88619291190d029a59ed4b3/src/opnsense/mvc/app/controllers/OPNsense/Core/Api/SystemController.php#L53-L57